plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/util/SynthesizedColumnHandler.java

```diff
@@ -238,4 +251,68 @@ public long getFileModificationTime()
             return modifiedTime;
         }
     }
+
+    /**
+     * Helper function to prefill BlockBuilders with values from PartitionKeys which are in the string type.
+     * This function handles the casting of String type the actual column type.
+     */
+    private static void appendToBuilder(Type type, Object value, BlockBuilder blockBuilder)
```

Comment:
Is the passed-in `value` always `String`-typed?

Reply:
Yes, always passed as String type. Will do.

Reply:
Ignore above comment, after checking, `Type type` is the target type that needs to be casted to, so, we still need it.

I will rename it to `targetType` to make it more intuitive.

Reply:
What I mean is: `Object value` -> `String value`, as the passed in value is always a String object regardless of the target type?